code-challenges.test.js

```diff
@@ -37,10 +76,44 @@ var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
+describe("capFirst", () => {
+    it("returns the input word with the first letter capitalized", () => {
+        //expect the output to equal the array but with all first letters capitalized
+        expect(capFirst(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(capFirst(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"])
+    })
+})
 
 
 // b) Create the function that makes the test pass.
 
+// create a test for the function of capitalizing first letter of a string
+describe("firstUpper", () => {
+    it("returns the input string but with the first letter capitalized", () => {
+        // the output should be the input but wwith the first letter capitalized
+        expect(firstUpper("vegan")).toEqual("Vegan")
+        expect(firstUpper("random")).toEqual("Random")
+    })
+
+})
+//Got red
+
+//create a function that capitalizes the first letter of a string using .toUpperCase
+const firstUpper = (string) => {
+    //Return the lettter at index(0) capitalized and the rest of the string concatenated 
+    return string.charAt(0).toUpperCase() + string.slice(1)
+}
+//Got Green
+
+// declare the function capFirst that capitalizes the first letter of each string in an array
+const capFirst = (array) => {
+    // return the input array using .map to iterate over the array and return a new array
+    return array.map(firstUpper)
+}
+//Got green, moving to refactor => is it possible to combine the two functions into one?
+//Declare function that takes the array in and then uses map combined with the return from firstUpper to output expected array.
+//I think anything I doto combine them will make it longer or harder to read
+
```

Comment:
Breaking the functions out is fine. But they can absolutely be combined into one, very readable, function.